main.js

```diff
@@ -118,4 +128,51 @@ document.addEventListener("DOMContentLoaded", function () {
         break;
     }
   }
+
+  // Edit a to-do item
+  function editTodoItem(item) {
+    const textSpan = item.querySelector(".text");
+    const newText = prompt("Edit task", textSpan.innerText);
+
+    if (newText !== null) {
+      textSpan.innerText = newText;
+      saveTasks();
+    }
+  }
+
+  // Removing and editing tasks
+  todoList.addEventListener("click", (event) => {
+    if (event.target.classList.contains("remove")) {
+      removeTodoItem(event.target.parentElement);
+    } else if (event.target.classList.contains("edit")) {
+      editTodoItem(event.target.parentElement.parentElement);
+    }
+  });
+
+  // Save tasks to localStorage
+  function saveTasks() {
+    const tasks = [];
+    todoList.querySelectorAll("li").forEach((item) => {
+      const text = item.querySelector(".text").innerText;
+      const isChecked = item.querySelector("input[type='checkbox']").checked;
+
+      tasks.push({ text, isChecked });
+    });
+
+    localStorage.setItem("tasks", JSON.stringify(tasks));
+  }
```

Comment:
The comment is still actual https://github.com/aminzoda/todo-app/pull/4#discussion_r1694242701

Reply:
Read more here https://css-tricks.com/build-a-state-management-system-with-vanilla-javascript/

Reply:
You can implement Flux architecture https://habr.com/ru/articles/249279/ which is used by React https://medium.com/@marina.kovalyova/flux-the-react-js-application-architecture-773f515d068d